### PR TITLE
add __enter__, __exit__ and __getattr__ to streamreader/writer

### PR DESCRIPTION
**Pull request by martindemello**
codecs.StreamReader and codecs.StreamWriter were missing a few methods.

### CODE REVIEW

File: stdlib/2and3/codecs.pyi [hunk 1]
```diff
@@ -132,6 +134,13 @@ class StreamWriter(Codec):
     def write(self, obj: _Decoded) -> None: ...
     def writelines(self, list: Iterable[_Decoded]) -> None: ...
     def reset(self) -> None: ...
+    def __enter__(self: _SW) -> _SW: ...
+    def __exit__(
+        self, typ: Optional[Type[BaseException]], exc: Optional[BaseException], tb: Optional[types.TracebackType]
+    ) -> bool: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/codecs.pyi]: Actually just returns None.

Comment (JelleZijlstra) [reply stdlib/2and3/codecs.pyi]: I haven't looked at them; can you check their implementations?

**Comment by martindemello** [comment]
ping @JelleZijlstra - github doesn't seem to have a good way to mark review changes complete

**Comment by JelleZijlstra** [comment]
Sorry for that, thanks for the ping!

**Pull request merged**